Add dispersion to NXmonochromator

lukaspie opened the pull request:
I only added a very simple definition (so that it can be used), this could potentially be enhanced in the future.

rettigl reviewed (approved):
LGTM. Only thing: It appears everything is defined both as wavelength and energy, so even if we don't need it, we could define both energy_dispersion and wavelength_dispersion.

lukaspie commented:
> LGTM. Only thing: It appears everything is defined both as wavelength and energy, so even if we don't need it, we could define both energy_dispersion and wavelength_dispersion.

I added `energy_dispersion `and `wavelength_dispersion `and removed `dispersion`.
